### PR TITLE
[#6638] SimpleQuery: Deprecate types, constants, and functions (main)

### PR DESCRIPTION
**Pull request by alanking**
Addresses #6638 (the rest of it)

Follow-up to #6657 

Just left a comment for the API number, but the rest has now received a deprecation attribute. Had to wrap some bits to ignore the deprecated declarations warnings that are now generated.

### CODE REVIEW

**Review by korydraughn — APPROVED** [review]
Ship it!

**Comment by korydraughn** [comment]
Let's wait for another approval before the pounds though.

File: server/api/src/rsSimpleQuery.cpp
Comment (trel) [inline server/api/src/rsSimpleQuery.cpp:94]: why does this one get a comment?  the others didn't...

Comment (alanking) [reply server/api/src/rsSimpleQuery.cpp:94]: I put a comment on this one because the `clang diagnostic push` occurs at the top of the file (there's a lot of deprecated stuff being used in here :p). The other `push`es and `pop`s were close enough to one another that they could be visually inspected very easily to understand what is happening.

That said, I'm happy to add comments to the others, or remove this comment

Comment (trel) [reply server/api/src/rsSimpleQuery.cpp:94]: all good, don't feel strongly - just noticed the difference.

Comment (alanking) [reply server/api/src/rsSimpleQuery.cpp:94]: In that case, I am going to leave things how they are. I'm hoping the next time we look at any of this is when SimpleQuery gets removed.

**Review by trel — APPROVED** [review]

**Pull request merged**
